src/ssl/bio.cc

```diff
@@ -554,6 +555,16 @@ Ssl::ServerBio::resumingSession()
     return parser_.resumingSession;
 }
 
+bool
+Ssl::ServerBio::tls1p3orLater() const
+{
+    // Parsing may not be completed or aborted because of an error
+    if (!Security::IsTlsProtocol(parser_.details->tlsSupportedVersion))
```

Comment:
Here, I think we should test whether we know the protocol, not whether the protocol is TLS or SSL.

* If we do not know the protocol, the method should return false -- your comment is accurate.
* If the protocol here is, say, HTTP, then it is a bug (leading to a desirable exception being thrown), rather than a regular/supported "return false" condition.

Reply:
Yes we can replace this check with a simple "if (!parser_.details->tlsSupportedVersion)"

Reply:
Done in 36cd9bb.